pkg/api/pkcs11ca_signing_cert.go

```diff
@@ -35,14 +38,31 @@ func FulcioCASigningCertHandler(subj *challenges.ChallengeResult, publicKey []by
 
 	rootID := []byte(viper.GetString("hsm-caroot-id"))
 
-	// get the existing root CA from the HSM
-	rootCA, err := p11Ctx.FindCertificate(rootID, nil, nil)
-	if err != nil {
-		return "", nil, err
+	// get the existing root CA from the HSM or from disk
+	var rootCA *x509.Certificate
+	if !viper.IsSet("aws_hsm_root_ca_path") {
+		rootCA, err = p11Ctx.FindCertificate(rootID, nil, nil)
+		if err != nil {
+			return "", nil, err
+		}
+	} else {
+		rootCaPath := filepath.Clean(viper.GetString("aws_hsm_root_ca_path"))
+		pubPEMData, err := os.ReadFile(rootCaPath)
+		if err != nil {
+			return "", nil, err
+		}
+		block, _ := pem.Decode(pubPEMData)
+		if block == nil || block.Type != "CERTIFICATE" {
+			log.Logger.Fatal("failed to decode PEM block containing certificate")
```

Comment:
the intent here is good, but would cause a dropped connection to the client. If we did this once at initialization, that address this issue as well